pkg/kubelet/kubelet.go

```diff
@@ -957,7 +959,17 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	}
 
 	// Start volume manager
-	go kl.volumeManager.Run(wait.NeverStop)
+	readyCh := make(chan bool, 1)
+	go kl.volumeManager.Run(readyCh, wait.NeverStop)
+	// Ready channel is set when all the sources are ready
+	go func() {
+		for {
+			if kl.sourcesReady.AllReady() {
+				readyCh <- true
+				break
```

Comment:
@saad-ali @jingxu97 @thockin Is it reasonable to write a func that starts up the volume manager for the kubelet and handles lines 962-972? Then we can comment the routine and explain its scope and purpose.